Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/CopyAndPasteController.java

```diff
@@ -344,4 +344,45 @@ public void paste(FolderType currentFolder) throws Exception {
       LOGGER.error(e.getMessage());
     }
   }
+  
+  public void drop(Set<Object> selections, int selectionCount, FolderType currentFolder) throws Exception {
+	  
+	   if (currentFolder == null) {
+	      Notification.error(Labels.getLabel("portal_failedFind_message"));
+	      return;
+	    }
+	    if (!ItemHelpers.isOwner(this.currentUser, currentFolder)) {
+	      Notification.error(Labels.getLabel("portal_onlyOwnerCanPasteToCurrent_message"));
+	      return;
+	    }
+	  
+		if (checkContext(selections, selectionCount, currentFolder)) {
+			try {
+				if (!checkImmediateOwnership(selections)) {
+					Notification.error(Labels.getLabel("portal_onlyOwnerCanCutItems}"));
+				} else if (ItemHelpers.isOwner(this.currentUser, currentFolder)) {
+					isCut = true;
+					updateSelectedItems(selections);
+				} else {
+					Notification.error(Labels.getLabel("portal_onlyOwnerCanCutFromCurrent_message"));
+				}
+			} catch (Exception e) {
+				Messagebox.show(Labels.getLabel("portal_failedCut_message"), "Apromore", Messagebox.OK,
+						Messagebox.ERROR);
+				LOGGER.error(e.getMessage());
+			}
+		}
+	   
+		try {
+			if (isCut) {
+				selectedTargetFolderId = currentFolder.getId();
+				moveSelectedItems();
+				Notification.info(Labels.getLabel("portal_itemsDropped_message"));
+				clearSelectedItems();
+			}
+		} catch (Exception e) {
+			Messagebox.show(Labels.getLabel("portal_failedPaste_message"), "Apromore", Messagebox.OK, Messagebox.ERROR);
+			LOGGER.error(e.getMessage());
```

Comment:
Again, it'd be better to have a stack trace, e.g. `LOGGER.error("Unable to perform Paste", e);`